pkg/registry/apis/provisioning/controller/repository.go

```diff
@@ -360,9 +361,11 @@ func (rc *RepositoryController) determineSyncStrategy(ctx context.Context, obj *
 }
 
 func (rc *RepositoryController) addSyncJob(ctx context.Context, obj *provisioning.Repository, syncOptions *provisioning.SyncJobOptions) error {
-	job, err := rc.jobs.Add(ctx, &provisioning.Job{
+	job, err := rc.jobs.Insert(ctx, &provisioning.Job{
 		ObjectMeta: v1.ObjectMeta{
 			Namespace: obj.Namespace,
+			// Sync and migrate jobs should never run at the same time. Hence, the name encapsulates them both (and the spec differentiates them).
+			Name: fmt.Sprintf("%s-syncmigrate", obj.GetName()),
```

Comment:
ummm. Seems like a hack. Shouldn't the queue deal with this and not trust the caller? 



Did you think about having a parent job field? The parent job which triggered it as migrate is a compound one.

Reply:
Having the store deal with the name seems like a better idea, yes. I'll push an update.